doc/architecture.rst

```diff
@@ -370,6 +370,38 @@ Example:
+
+Encrypting files locally means the server has no access to them. This brings with it a number of limitations:
+
+* E2EE files can not be accessed or previewed through the web interface
+* E2EE files can not be edited with Online Office solutions
+* E2EE files can not be shared with a public link
+* E2EE files can not be searched, tagged, commented on and have no versioning or trash bin
+* E2EE files can not be accessed in other Nextcloud Apps. This means they have no chat sidebar, can not be attached to emails or deck cards, shared in Talk rooms and so on
+* E2EE results in slower syncing of file and works poorly or not at all with large files and large quantities of files
+
+These limitations are fundamental to how securely implemented end-to-end encryption works. We realize there are some solutions that call their technology 'end-to-end encryption' but with browser access. Reality is that offering browser access to end-to-end encrypted files would essentially negate any of the benefits of end-to-end encryption. Serving a file in the browser means the server needs to be able to read the files. But if the server can read the files, administrator or a malicious attacker who gained access to the server, can too. Decrypting the file in the browser does not solve this security risk in the least, as the javascript code that would be needed to decrypt the file comes FROM the server, and of course a compromised server would simply send modified javascript code which sends a copy of the encryption keys to the attacker without anybody noticing. See for more details our blog about the `threat model for the encryption solutions in Nextcloud`_ and our `webpage about End-to-end Encryption`_.
+
+The E2EE design of Nextcloud allows for sharing on a per-folder level to individual users (not groups), but, as of early 2021, this feature is still on the road map for implementation in the clients.
+
+Due to all these limitations that are inherent to true end-to-end encryption, it is only recommended for a small subset of files, in just a small number of folders. Encrypting your entire sync folder is likely to result in poor performance and sync errors and if you do not trust your server at all, Nextcloud is perhaps not the right solution for your use case. You might instead want to use encrypted archives or another solution.
+
+.. note::
+    * End-to-end Encryption does not work with Virtual Files (VFS). Folders with E2EE have to be made available offline to access the files, they can not be retrieved on demand.
```

Comment:
Note that this limitation is only true on Windows. Granted it's only the backend in production, but I suspect if this limitation is not made more precise now it'll get overlooked when other backends come up online.

Reply:
Also I realize after posting: they do work in a way. It's just the implicit download which doesn't work on Windows. This is what you're saying, but I suspect most people will read only the first sentence. :-)

Reply:
You are 100% correct, saying it doesn't work really is too negative. Thanks!